src/implementation.md

```diff
@@ -54,7 +54,10 @@ Implementation will use a single `boost::asio`-loop based approach without extra
 
 All platform-specific code will be abstracted out. That would allow us to compile the same C++ code both to native binaries and to a WASM module.
 
-Emscripten will be used to compile C++ code into WASM module.
+We will use [binaryen](https://github.com/WebAssembly/binaryen) with Emscripten to create the WASM module. Binaryen is a compiler and toolchain infrastructure library for WebAssembly written in C++. Reasons behind this choice:
```

Comment:
I would change the wording and maybe separate Emscripten from Bynaryen.
Emscripten is a toolset to compile light client to WebAssembly itself .
While Bynaryen is a library to run WebAssembly *inside* light client